### PR TITLE
FAI-12885: Zephyr Scale Source Streams stubs

### PR DESCRIPTION
**Pull request by chalenge**
## Description

- Zephyr Scale Source Streams stubs

## Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

### CODE REVIEW

File: sources/zephyr-source/package.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      "transform": {
+        "\\.[jt]sx?$": [
+          "ts-jest",
+          {
+            "tsconfig": "test/tsconfig.json"
+          }
+        ]
+      }
+    }
+  }
```
Comment (ypc-faros) [inline sources/zephyr-source/package.json:56]: nit: new line

File: sources/zephyr-source/resources/spec.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        "api_max_retries": {
+          "order": 6,
+          "type": "integer",
+          "title": "Max Number of Retries",
+          "description": "The max number of retries before giving up on retrying requests to the Zephyr API.",
+          "default": 3
+        }
+      }
+    }
+  }
```
Comment (ypc-faros) [inline sources/zephyr-source/resources/spec.json:55]: nit: new line

File: sources/zephyr-source/resources/spec.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
+{
+    "documentationUrl": "https://docs.faros.ai",
+    "connectionSpecification": {
+      "$schema": "http://json-schema.org/draft-07/schema#",
+      "title": "Zephyr Scale Configuration Spec",
+      "type": "object",
+      "additionalProperties": true,
+      "properties": {
+        "access_token": {
```
Comment (ypc-faros) [inline sources/zephyr-source/resources/spec.json:9]: Any of these `required`?

File: sources/zephyr-source/src/streams/common.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    this.projects = config.projects;
+  }
+
+  // Most of the streams will have the same primary key
+  get primaryKey(): StreamKey {
+    return ['id'];
+  }
+
+  async *streamSlices(): AsyncGenerator<StreamSlice> {
+    this.logger.info('streamSlices');
```
Comment (ypc-faros) [inline sources/zephyr-source/src/streams/common.ts:35]: Remove these logs :)

File: sources/zephyr-source/resources/spec.json [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        "projects": {
+          "order": 2,
+          "type": "array",
+          "title": "Project Keys",
+          "description": "List of Jira project keys to sync.",
+          "items": {
+            "type": "string"
+          }
+        },
+        "cutoff_days": {
```
Comment (ypc-faros) [inline sources/zephyr-source/resources/spec.json:25]: We don't seem to do anything with the `cutoff` (yet?)

File: sources/zephyr-source/test_files/full_configured_catalog.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+          "json_schema": {},
+          "supported_sync_modes": [
+            "full_refresh"
+          ]
+        },
+        "sync_mode": "full_refresh",
+        "destination_sync_mode": "overwrite"
+      }
+  ]
+}
```
Comment (ypc-faros) [inline sources/zephyr-source/test_files/full_configured_catalog.json:37]: nit: new line

**Review by ypc-faros — APPROVED** [review]

File: sources/zephyr-source/resources/spec.json (hunk 3 above)
Comment (chalenge) [reply sources/zephyr-source/resources/spec.json:9]: Updated spec

File: sources/zephyr-source/resources/spec.json (hunk 5 above)
Comment (chalenge) [reply sources/zephyr-source/resources/spec.json:25]: Nope, removed

**Pull request merged**